src/client.rs

```diff
@@ -38,14 +35,20 @@ pub struct Body<B: IntoBuf> {
     inner: proto::StreamRef<B::Buf, Peer>,
 }
 
+/// Configure a Client from the defaults.
+#[derive(Debug, Default)]
+pub struct Config {
```

Comment:
I would consider renaming this to `Builder` as this looks more like a builder pattern.